pulsar-client/src/main/java/org/apache/pulsar/client/impl/TypedMessageBuilderImpl.java

```diff
@@ -100,7 +100,6 @@ public MessageId send() throws PulsarClientException {
         CompletableFuture<MessageId> sendFuture;
         if (txn != null) {
             sendFuture = producer.internalSendWithTxnAsync(message, txn);
-            txn.registerSendOp(sendFuture);
```

Comment:
should this be removed?

Reply:
Yes, I moved the call into `ProducerImpl::internalSendWithTxnAsync`.